### PR TITLE
Support dynamic domains in mod_adhoc

### PR DESCRIPTION
**Pull request by Premwoik**
This PR adds support for dynamic domains in mod_adhoc and adapts tests.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71054) / small_tests / 502642a26160d4bd17d865599a59bdc2c9031794
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3252/71054/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3252/71054/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-09_07.33.28/lib.mongooseim.logs/run.2021-09-09_07.33.28/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71113) / small_tests / 1dc035e558fbc11a8688e4170e5441485c704433
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3252/71113/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3252/71113/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-09_11.04.02/lib.mongooseim.logs/run.2021-09-09_11.04.03/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71142) / small_tests / 16baba98cbe4ec46ab888eb4384c951d129b45ce
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3252/71142/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3252/71142/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-09_12.00.30/lib.mongooseim.logs/run.2021-09-09_12.00.30/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71191) / small_tests / a66303ce6410c039be194bf965600e99d4505300
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3252/71191/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3252/71191/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-09_12.55.33/lib.mongooseim.logs/run.2021-09-09_12.55.34/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71216) / small_tests / 10473f51ee5536bfa10d217e99bf1b666a057328
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3252/71216/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3252/71216/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-09_13.02.55/lib.mongooseim.logs/run.2021-09-09_13.02.56/suite.log.html)

File: src/mod_adhoc.erl [hunk 1]
```diff
@@ -31,11 +31,15 @@
 -type command_hook_acc() :: {error, exml:element()} | exml:element() | ignore | empty.
 -export_type([command_hook_acc/0]).
 
+%% Gen_mod callbacks
```
Comment (vkatsuba) [inline src/mod_adhoc.erl:34]:
```suggestion
%% gen_mod callbacks
```

**Review by chrzaszcz — APPROVED** [review]
Looks good!

File: src/mod_adhoc.erl (hunk 1 above)
Comment (chrzaszcz) [reply src/mod_adhoc.erl:34]: This is the same comment as in `mod_last` so I decided we could merge it like it is. I could fix both with some upcoming PR.

**Pull request merged**